python/lsst/jointcal/jointcal.py

```diff
@@ -386,27 +388,47 @@ def writeModel(model, filename, log):
     log.info("Wrote %s to file: %s", model, filename)
 
 
-class JointcalTask(pipeBase.CmdLineTask):
-    """Jointly astrometrically and photometrically calibrate a group of images."""
+@dataclasses.dataclass
+class JointcalInputData:
```

Comment:
I see where you're going here with making the common dataclass for gen2 and gen3 to normalize things.  However, I'm not sure if this will actually work (which is maybe part of the point of the new ticket?) because (a) the gen3 version (at least!) should be using the parquet files; (b) by reading all the catalogs first and not cutting them down as you read them this leads to significant memory overhead.  But having a common dataclass in general does make sense.

Reply:
a) We'll see how this looks when I write the Parquet version. This class may not be as helpful then.

b) The gen2 code is still only reading one catalog/image at a time, so no memory pressure there. The dataclass is just to help separate the "read input" code (which is different in gen2/gen3) from the "create internal jointcal representation" code (which is middleware agnostic).